Tests/Tests/STPPaymentContextTest.m

```diff
@@ -0,0 +1,97 @@
+//
+//  STPPaymentContextTest.m
+//  Stripe
+//
+//  Created by Brian Dorfman on 8/1/16.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "STPAPIClient.h"
+#import "STPPaymentContext.h"
+
+@interface STPPaymentContext (Testing)
+- (PKPaymentRequest *)buildPaymentRequest;
+@end
+
+@interface TestSTPBackendAPIAdapter: NSObject <STPBackendAPIAdapter>
+@end
+
+@implementation TestSTPBackendAPIAdapter
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-parameter"
+- (void)retrieveCustomer:(STPCustomerCompletionBlock)completion {
+
+}
+
+- (void)attachSourceToCustomer:(id<STPSource>)source completion:(STPErrorBlock)completion {
+    
+}
+
+- (void)selectDefaultCustomerSource:(id<STPSource>)source completion:(STPErrorBlock)completion {
+    
+}
+#pragma clang diagnostic pop
+
+@end
+
+@interface STPPaymentContextTest : XCTestCase
+
+@end
+
+@implementation STPPaymentContextTest
+
+- (void)setUp {
+    [super setUp];
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+    [Stripe setDefaultPublishableKey:@"test"];
+    [STPPaymentConfiguration sharedConfiguration].appleMerchantIdentifier = @"testMerchantId";
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    [super tearDown];
+}
+
+- (void)testPKPaymentTotalAmount {
```

Comment:
We usually name test cases after the method being tested, e.g.

```
testBuildPaymentRequest_totalAmount
testBuildPaymentRequest_USDDefault
etc
```

I also like naming the subject under test (STPPaymentContext) `sut`, but that's mostly personal preference.